internal/internal.go

```diff
@@ -612,39 +620,121 @@ func getParsedPathAndQuery(req *ext_authz.CheckRequest) ([]interface{}, map[stri
 	return parsedPathInterface, parsedQueryInterface, nil
 }
 
-func getParsedBody(req *ext_authz.CheckRequest) (interface{}, bool, error) {
-	body := req.GetAttributes().GetRequest().GetHttp().GetBody()
+func getParsedBody(req *ext_authz.CheckRequest, parsedPath []interface{}, protoDescriptor string) (interface{}, bool, error) {
 	headers := req.GetAttributes().GetRequest().GetHttp().GetHeaders()
 
-	if body == "" {
-		return nil, false, nil
-	}
-
-	if val, ok := headers["content-length"]; ok {
-		cl, err := strconv.ParseInt(val, 10, 64)
-		if err != nil {
-			return nil, false, err
-		}
-
-		if cl != -1 && cl > int64(len(body)) {
-			return nil, true, nil
-		}
-	}
-
 	var data interface{}
 
 	if val, ok := headers["content-type"]; ok {
 		if strings.Contains(val, "application/json") {
+
+			body := req.GetAttributes().GetRequest().GetHttp().GetBody()
+
+			if body == "" {
+				return nil, false, nil
+			}
+
+			if val, ok := headers["content-length"]; ok {
+				cl, err := strconv.ParseInt(val, 10, 64)
+				if err != nil {
+					return nil, false, err
+				}
+				if cl != -1 && cl > int64(len(body)) {
+					return nil, true, nil
+				}
+			}
+
 			err := util.Unmarshal([]byte(body), &data)
 			if err != nil {
 				return nil, false, err
 			}
+		} else if strings.Contains(val, "application/grpc") {
+			rawbody := req.GetAttributes().GetRequest().GetHttp().GetRawBody()
+
+			if len(rawbody) <= 0 {
+				return nil, false, fmt.Errorf("invalid raw body")
+			}
+			if len(parsedPath) <= 1 {
+				return nil, false, fmt.Errorf("invalid parsed path")
+			}
+
+			if protoDescriptor == "" {
+				return nil, false, fmt.Errorf("ProtoDescriptor path not defined")
+			}
+
+			err := getGRPCBody(rawbody, parsedPath, &data, protoDescriptor)
+			if err != nil {
+				return nil, false, err
+			}
 		}
 	}
 
 	return data, false, nil
 }
 
+func getGRPCBody(in []byte, parsedPath []interface{}, data interface{}, protoDescriptor string) error {
+
+	// the first 5 bytes are part of gRPC framing. We need to remove them to be able to parse
+	//https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md
+	in = in[5:]
+
+	bytes, err := ioutil.ReadFile(protoDescriptor) //grpcprotoset/data.pb")
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+
+	var fileSet descriptor.FileDescriptorSet
+	if err := proto.Unmarshal(bytes, &fileSet); err != nil {
+		return err
+	}
+
+	fd, err := desc.CreateFileDescriptorFromSet(&fileSet)
+	if err != nil {
+		return err
+	}
+
+	inputType := ""
+	packageName := fd.GetPackage()
+	pathServiceRaw := string(parsedPath[0].(string))
```

Comment:
Hmm I think this would be enough:
```suggestion
	pathServiceRaw := parsedPath[0].(string)
```